File: plugins/maximum.py
import re
from plugin import Plugin


class MaximumPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'最大\s+(.+)', line)

        if not match:
            return None

        values = match.group(1)

        return f"print(max({values}))"

File: web/plugins/maximum.js
import { WebPlugin } from "../engine/plugin.js";

export class MaximumPlugin extends WebPlugin {
    constructor() {
        super("最大");
    }

    async execute(line, context) {
        const expression = line
            .replace(/^最大\s*/, "");

        const values = context.parser
            .splitArguments(expression)
            .map(value =>
                Number(
                    context.parser.parseValue(
                        value,
                        context.runtime
                    )
                )
            );

        context.runtime.write(
            Math.max(...values)
        );
    }
}